tensorflow_io/bigquery/python/ops/bigquery_api.py

```diff
@@ -115,21 +126,40 @@ def read_session(
         if not dataset_id:
             raise ValueError("`dataset_id` must be a set")
 
-        if not isinstance(selected_fields, list):
-            raise ValueError("`selected_fields` must be a list")
-        if not selected_fields:
-            raise ValueError("`selected_fields` must be a set")
-
-        if not isinstance(output_types, list):
-            raise ValueError("`output_types` must be a list")
-        if output_types and len(output_types) != len(selected_fields):
-            raise ValueError(
-                "lengths of `output_types` must be a same as the "
-                "length of `selected_fields`"
-            )
-
-        if not output_types:
-            output_types = [dtypes.string] * len(selected_fields)
+        if isinstance(selected_fields, list):
+            if not isinstance(output_types, list):
+                raise ValueError(
+                    "`output_types` must be a list if selected_fields is list"
+                )
+            if output_types and len(output_types) != len(selected_fields):
+                raise ValueError(
+                    "lengths of `output_types` must be a same as the "
+                    "length of `selected_fields`"
+                )
+            if not output_types:
+                output_types = [dtypes.string] * len(selected_fields)
+            # Repeated field is not supported if selected_fields is list
+            selected_fields_repeated = [False] * len(selected_fields)
+
+        elif isinstance(selected_fields, dict):
+            _selected_fields = []
+            selected_fields_repeated = []
+            output_types = []
+            for field in selected_fields:
+                _selected_fields.append(field)
+                mode = selected_fields[field].get("mode", "nullable")
+                if mode == "repeated":
```

Comment:
could you add a FieldMode enum similarly to DataFormat, so strings are not hardcoded.

Reply:
added.